bb: add page (#9500)
tldr-pages/tldr: pages/common/bb.md

# bb

> Native Clojure interpreter for scripting.
> More information: <https://book.babashka.org/#usage>.

- [e]valuate an expression:

`bb -e "(+ 1 2 3)"`

- Evaluate a script [f]ile:

`bb -f {{path/to/script.clj}}`

> - Bind input to a lazy seq of lines from stdin:

Reviewer: ```suggestion
- Bind input to a sequence of lines from stdin:
```
Is the word lazy necessary?

`printf "first\nsecond" | bb -i "(map clojure.string/capitalize *input*)"`

> - Bind input to a lazy seq of EDN values from stdin:

Reviewer: ```suggestion
- Bind input to a sequence of EDN values from stdin:
```

> - Bind input to a sequence of EDN values from stdin:

Reviewer 2: What is END values?

Author: extensible data notation. It's the data format used to write clojure: https://github.com/edn-format/edn

Reviewer: > extensible data notation. It's the data format used to write clojure: https://github.com/edn-format/edn

How about adding it too in the command description?

```suggestion
- Bind input to a sequence of EDN(Extensible Data Notation) values from stdin:
```

# bb

> > Scripting interpreter.

Reviewer: ```suggestion
```
I don't think this line is necessary as you are already mentioning it below.

> Native Clojure interpreter for scripting.

> > <https://book.babashka.org/#usage>.

Reviewer: ```suggestion
> More information: <https://book.babashka.org/#usage>.
```

> - [e]valuate expression:

Reviewer: ```suggestion
- [e]valuate an expression:
```
Would this be better?

`bb -e "(+ 1 2 3)"`

> - Evaluate a [f]ile:

Reviewer: ```suggestion
- Evaluate a script [f]ile:
```

> `bb -f script.clj`

Reviewer: ```suggestion
`bb -f {{path/to/script.clj}}`
```
This is the format we use for paths.

> - Bind *input* to a lazy seq of lines from stdin:

Reviewer: ```suggestion
- Bind input to a lazy seq of lines from stdin:
```
Italics wouldn't work properly with some of our clients.

`printf "first\nsecond" | bb -i "(map clojure.string/capitalize *input*)"`

> - Bind *input* to a lazy seq of EDN values from stdin:

Reviewer: ```suggestion
- Bind input to a lazy seq of EDN values from stdin:
```
